cmplibrary/src/main/java/com/sourcepoint/cmplibrary/data/network/model/optimized/сonsentStatus/USNatCS.kt

```diff
@@ -0,0 +1,152 @@
+package com.sourcepoint.cmplibrary.data.network.model.optimized.сonsentStatus
+
+import com.sourcepoint.cmplibrary.data.network.converter.JsonConverter
+import com.sourcepoint.cmplibrary.data.network.converter.JsonConverterImpl
+import com.sourcepoint.cmplibrary.data.network.converter.JsonMapSerializer
+import com.sourcepoint.cmplibrary.data.network.converter.converter
+import com.sourcepoint.cmplibrary.data.network.model.optimized.CampaignMessage
+import com.sourcepoint.cmplibrary.data.network.model.optimized.MessageMetaData
+import com.sourcepoint.cmplibrary.data.network.model.optimized.USNatConsentStatus
+import com.sourcepoint.cmplibrary.exception.CampaignType
+import com.sourcepoint.cmplibrary.model.exposed.ConsentableImpl
+import com.sourcepoint.cmplibrary.model.exposed.UsNatConsentInternal
+import com.sourcepoint.cmplibrary.util.extensions.toMapOfAny
+import com.sourcepoint.mobile_core.models.consents.USNatConsent
+import com.sourcepoint.mobile_core.network.responses.ChoiceAllResponse
+import com.sourcepoint.mobile_core.network.responses.USNatChoiceResponse
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.JsonObject
+import com.sourcepoint.cmplibrary.core.getOrNull
+import com.sourcepoint.cmplibrary.util.check
+import kotlinx.serialization.encodeToString
+
+
+@Serializable
+data class USNatCS(
+    val applies: Boolean? = null,
+    @SerialName("consentStatus") val consentStatus: USNatConsentStatus? = null,
+    @SerialName("consentStrings") val consentStrings: List<ConsentString>? = null,
+    @SerialName("dateCreated") override var dateCreated: String? = null,
+    @SerialName("uuid") var uuid: String? = null,
+    @SerialName("webConsentPayload") val webConsentPayload: JsonObject? = null,
+    @SerialName("message") override val message: JsonElement? = null,
+    @SerialName("GPPData") @Serializable(with = JsonMapSerializer::class) val gppData: Map<String, JsonElement>? = emptyMap(),
+    @SerialName("messageMetaData") override val messageMetaData: MessageMetaData? = null,
+    @SerialName("type") override val type: CampaignType = CampaignType.USNAT,
+    @SerialName("url") override val url: String? = null,
+    @SerialName("expirationDate") override val expirationDate: String? = null,
+    val userConsents: UserConsents? = null
+) : CampaignMessage {
+    val vendors: List<ConsentableImpl>
+        get() { return userConsents?.vendors ?: emptyList() }
```

Comment:
I believe `get()` can be inlined as well:
```
   get() = userConsents?.vendors ?: emptyList()
```

Not sure though.

Reply:
I'll check

Reply:
Yes, it can be done this way! I've done this code review fix in the https://github.com/SourcePointUSA/android-cmp-app/pull/844